docs/development_checklist_integration.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Checklist for creating an integration"
+sidebar_label: Integration Checklist
+---
+
+A checklist of things to do when you're adding a new integration.
+
+> Not all existing code follow the requirements in this checklist. This cannot be used as a reason to not follow them!
+
+### 0. Common
+
+ 1. Follow our [Style guidelines](development_guidelines.md)
+ 2. Use existing constants from [`const.py`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/const.py)
+    * Only add new constants to `const.py` if they are widely used. Otherwise keep them at the integration level
+
+### 1. Requirements
+
+ 1. Requirements have been added to [`manifest.json`](creating_integration_manifest.md). The `REQUIREMENTS` constant is deprecated.
+ 2. Requirement version should be pinned: `"requirements": ['phue==0.8.1']`
+ 3. We no longer want requirements hosted on GitHub. Please upload to PyPi.
+
+### 2. Configuration
+
+ 1. Voluptuous schema present for [configuration validation](development_validation.md)
+ 2. Default parameters specified in voluptuous schema, not in `setup(…)`
+ 3. Schema using as many generic config keys as possible from `homeassistant.const`
+ 4. If your integration has platforms, define a `PLATFORM_SCHEMA` instead of a `CONFIG_SCHEMA`.
+ 5. If using a `PLATFORM_SCHEMA` to be used with `EntityComponent`, import base from `homeassistant.helpers.config_validation`
+ 6. Never depend on users adding things to `customize` to configure behavior inside your integration.
+
+### 3. Integration/platform communication
+
+ 1. You can share data with your platforms by leveraging `hass.data[DOMAIN]`.
+ 2. If the integration fetches data that causes its related platform entities to update, you can notify them using the dispatcher code in `homeassistant.helpers.dispatcher`.
```

Comment:
If the integration fetches data in the component module that causes the integration platform entities...

Reply:
Changed!